### PR TITLE
Introduction section on commands and subsystems

### PR DESCRIPTION
**Pull request by binnur**
@riyadth take a read and delete branch after merging. thanks!

### CODE REVIEW

File: commands_subsystems/README.md [hunk 1]
```diff
@@ -1,25 +1,62 @@
...
+
+### Subsystem design steps
+1. Describe your subsystem's behavior and how it transitions. Behaviors are activities that we can observe externally: _move forward_, _turn_, _stop_... Think about what behaviors can run concurrently and what requires a sole attention: _can you chew gum and brush teeth at the same time?_
+2. Brainstorm things your subsystem needs to do: _move forward some distance_, _turn some degrees_, _adjust speed_...
+3. Identify internal and/or external data it needs to do its job accurately and correctly: _sensor info on distance moved_, _current speed_, _current heading_...
+
+
+## Commands
+Commands enable us to operate our robot as a system. Commands are small chunks of tasks from different subsystems that we can chain together to perform more complex operations. As such, Commands can be tested individually. Each subsystem can also have a default command, which will run when no other running commands require that subsystem.
+- Move foreward 15 ft while lowering the arm and shoot
```
Comment (riyadth) [inline commands_subsystems/README.md]: 'foreward' is misspelled. 'forward' is correct

File: commands_subsystems/README.md [hunk 2]
```diff
@@ -1,25 +1,62 @@
...
+import edu.wpi.first.wpilibj.command.Command;
+import edu.wpi.first.wpilibj.command.CommandGroup
+import edu.wpi.first.wpilibj.command.Scheduler;
+
+public class IntakeLauncher extends Subsystem {
+  ...
+}
+```
+
+Commands are executed sequentially, and needs to be scheduled. They can be tied to autonomous modes, operator interface buttons, or SmartDashboard virtual buttons for scheduling.
```
Comment (riyadth) [inline commands_subsystems/README.md]: 'needs' should be 'need'

File: commands_subsystems/README.md [hunk 3]
```diff
@@ -1,25 +1,62 @@
...
+
+Commands can also be grouped together allowing the ability to run commands concurrently. Once the group is created, similar to Commands, it can be tied to a single entry to executing all the commands in this group. A command group finishes when all the commands started in that group finish.
+
+> Important: Commands require subsystem(s) to be present.  Command groups automatically require all the subsystems for each of the commands in the group. This puts additional constraints on your subsystem design: _should you combine two similarly related subsystems as one?_ Make sure you think through the operational needs of the system as that dependency may complicate future commands.
+
+
+## Scheduler
+Scheduler is responsible for scheduling Commands. Although many commands maybe running virtually at the same time, robot code is single threaded, i.e. there is only a single main robot thread. As a note, PID loops are self-contained threads that does not require scheduler.
```
Comment (riyadth) [inline commands_subsystems/README.md]: PID in the RoboRio (via WPIlib) is part of the same thread as all other execution. CANTalons are different CPUs that do their own thing.

File: commands_subsystems/README.md [hunk 4]
```diff
@@ -1,25 +1,62 @@
...
+
+Every 20ms (when driver station update takes place), Scheduler walks through its table of Commands to determine what commands need to be scheduled or marked as completed. For this, Scheduler calls each command's _execute()_ followed by _isFinished()_ methods. If _isFinished()_ is true, command is removed from the scheduling queue.
```
Comment (riyadth) [inline commands_subsystems/README.md]: ', the scheduler walks through...', not ', Scheduler walks through'. It's not a proper noun, so no upper casing, and it needs a 'the'. Same later in the sentence.

File: commands_subsystems/README.md [hunk 5]
```diff
@@ -1,25 +1,62 @@
...
+
+![Inside Scheduler's run method](images/scheduler-run.png)
+
+If a command is interruptable, scheduler will interrupt the running command to schedule the new one: pushing elevator down button will interrupt the elevator move up command.
```
Comment (riyadth) [inline commands_subsystems/README.md]: 'the scheduler'

**Review by riyadth — CHANGES_REQUESTED** [review]
Fix small changes and I will re-review.

File: commands_subsystems/README.md (hunk 1 above)
Comment (binnur) [reply commands_subsystems/README.md]: done!

File: commands_subsystems/README.md (hunk 2 above)
Comment (binnur) [reply commands_subsystems/README.md]: done!

File: commands_subsystems/README.md (hunk 3 above)
Comment (binnur) [reply commands_subsystems/README.md]: clarified.

File: commands_subsystems/README.md (hunk 4 above)
Comment (binnur) [reply commands_subsystems/README.md]: corrected...

File: commands_subsystems/README.md (hunk 5 above)
Comment (binnur) [reply commands_subsystems/README.md]: yup

**Pull request merged**